src/Compilers/CSharp/Test/Semantic/Diagnostics/GetDiagnosticsTests.cs

```diff
@@ -271,9 +271,13 @@ private static bool DequeueCompilationEvents(AsyncQueue<CompilationEvent> eventQ
                     {
                         var symbol = symbolDeclaredEvent.Symbol;
                         var added = declaredSymbolNames.Add(symbol.Name);
-                        Assert.True(added, "Unexpected multiple symbol declared events for symbol " + symbol);
-                        var method = symbol as Symbols.MethodSymbol;
-                        Assert.Null(method?.PartialDefinitionPart); // we should never get a partial method's implementation part
+                        if (!added)
+                        {
+                            var method = (Symbols.MethodSymbol)symbol;
```

Comment:
>(Symbols.MethodSymbol)symbol [](start = 41, length = 28)

There was no response to the question in relation to the safety of this cast. Is it safe to assume that it will not fail.

Reply:
if it fails, then it is a bug. only partial method can have added == false.

Reply:
@AlekseyTs ping.

Reply:
> only partial method can have added == false.

I am not sure why you are making this assumption. Type members can have conflicting names, overloaded methods can have the same name and they don't have to be partial.

Reply:
@AlekseyTs existing code checks that added was never false. now it can be false only if it is partial method symbol. if I change it differently, than I am changing semantic of this code more than what I changed.

Reply:
Existing code was asserting some expectations but it wouldn't cause a crash if the expectations are false. It looks like the new code is going to crash if the expectations are not met. If you want relax the assert condition, that is fine, but I would do that in way that wouldn't cause a crash instead of causing the assert failure.

Reply:
sure. I dont think it will crash though. it will fail with invalid cast exception.